activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1749,7 +1749,7 @@ def build_join_buckets
         end
 
         joins.each_with_index do |join, i|
-          joins[i] = Arel::Nodes::StringJoin.new(Arel.sql(join.strip)) if join.is_a?(String)
+          joins[i] = Arel::Nodes::StringJoin.new(Arel.sql(join.strip, retryable: true)) if join.is_a?(String)
```

Comment:
Again is this definitely safe?

Reply:
Thanks, yeah this can be user supplied! Will remove.